src/core/thread/address_resolver.hpp

```diff
@@ -132,28 +132,17 @@ class AddressResolver : public InstanceLocator, private NonCopyable
     void Remove(const Ip6::Address &aEid);
 
     /**
-     * This method updates an existing cache entry for the EID.
-     *
-     * @param[in]  aEid               A reference to the EID.
-     * @param[in]  aRloc16            The RLOC16 corresponding to @p aEid.
-     *
-     * @retval kErrorNone          Successfully updates an existing cache entry.
-     * @retval kErrorNotFound      No cache entry with @p aEid.
-     *
-     */
-    Error UpdateCacheEntry(const Ip6::Address &aEid, Mac::ShortAddress aRloc16);
-
-    /**
-     * This method adds a snooped cache entry for a given EID.
+     * This method updates an existing entry or adds a snooped cache entry for a given EID.
      *
      * The method is intended to add an entry for snoop optimization (inspection of a received message to create a
      * cache entry mapping an EID to a RLOC).
      *
-     * @param[in]  aEid               A reference to the EID.
-     * @param[in]  aRloc16            The RLOC16 corresponding to @p aEid.
+     * @param[in] aEid             A reference to the EID.
+     * @param[in] aRloc16          The RLOC16 corresponding to @p aEid.
+     * @param[in] aDest            The short MAC address destination of the received snooped message.
      *
      */
-    void AddSnoopedCacheEntry(const Ip6::Address &aEid, Mac::ShortAddress aRloc16);
+    void UpdateSnoopedCacheEntry(const Ip6::Address &aEid, Mac::ShortAddress aRloc16, Mac::ShortAddress aDest);
```

Comment:
```suggestion
    void Snoop(const Ip6::Address &aEid, Mac::ShortAddress aRloc16, Mac::ShortAddress aDest);
```
nit

Reply:
I still like `UpdateSnoopedEntry()` as the method name over `Snoop` since the `AddressResolver` is not really the one doing the "snooping" :)... it is only updating/adding a cache entry.